src/provider/handler/mappers/postMapper.js

```diff
@@ -1,36 +1,50 @@
-export function mapPost(post) {
-  let result = {
-    type: {
-      value: 'article'
-    },
-    media_group: [],
-    extensions: {},
-    content: {},
-    link: {}
-  };
-
-  result.id = post.id;
-  result.title = (post.title && post.title.rendered)?post.title.rendered:'';
-  result.publish = post.date;
+export function mapPost(mediaItems) {
+  return function(post) {
+    const result = {
+      type: {
+        value: 'article'
+      },
+      id: post.id,
+      title: post.title && post.title.rendered ? post.title.rendered : '',
+      publish: post.date,
+      media_group: [],
+      extensions: {},
+      content: {},
+      link: {}
+    };
 
-  //the post's link that will be used when a user shares the post
-  result.link = {
-    type: 'text/html',
-    rel: 'alternate',
-    href: post.link
-  };
+    //the post's link that will be used when a user shares the post
+    result.link = {
+      type: 'text/html',
+      rel: 'alternate',
+      href: post.link
+    };
 
-  //article's media id, to be fetched later
-  if (post.featured_media) {
-    result.featured_media = post.featured_media;
-  }
+    //if we can find the post's media id then let's add it's url to our media_group
+    mediaItems.forEach(mediaItem => {
+      if (
+        mediaItem.id === post.featured_media &&
+        result.media_group.length === 0
+      ) {
+        result.media_group.push({
+          type: 'image',
+          media_item: [
+            {
+              src: mediaItem.image,
+              key: 'image_base'
+            }
+          ]
+        });
+      }
+    });
 
-  //adding the post's content as an escaped string
-  if (post.content.rendered) {
-    result.content = {
-      html: encodeURIComponent(post.content.rendered)
-    };
-  }
+    //adding the post's content as an escaped string
+    if (post.content.rendered) {
+      result.content = {
+        html: encodeURIComponent(post.content.rendered)
```

Comment:
is this working ? I'm not sure we decode URIs when calling children links

Reply:
I thought that's what the documentation said about article.html - so I should just return the html as it is?

Reply:
Ok, now I understand.

HTML content may be escaped somehow - we will need to check with native developers, but I doubt it is done with the function that escapes special characters from url parameters